parens_2.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+
+import sys
+
+
+def _parens(prefix, left, right):
+    if left == 0 and right == 0: sys.stdout.write(prefix)
+    if left > 0: _parens(''.join([prefix, '(']), left-1, right)
```

Comment:
And when I restored the print() to get the same functionality, I get 25 seconds.